L1Trigger/TrackFindingTracklet/interface/DiskProjection.h

```diff
@@ -0,0 +1,168 @@
+#ifndef L1Trigger_TrackFindingTracklet_interface_DiskProjection_h
+#define L1Trigger_TrackFindingTracklet_interface_DiskProjection_h
+
+#include "L1Trigger/TrackFindingTracklet/interface/FPGAWord.h"
+
+#include <cassert>
+
+namespace Trklet {
+
+  class Settings;
+
+  class DiskProjection {
+  public:
+    DiskProjection() { valid_ = false; }
+
+    void init(const Settings* settings,
```

Comment:
We have an array of projections for each layer/disk. If the projection misses the layer it will not be valid. I guess the alternative would be to have a pointer that was null if not valid. But not clear that this would be any better.

Reply:
yes, having a pointer that is null if not initialized is better. partially initialized objects should be avoided. see https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization

Reply:
Even if this means creating and deleting a large number of these objects every event?

Reply:
No, anything that increases the number of allocations and deallocations should be avoided.

Reply:
Yes, I agree it's generally better to avoid a large number of allocations. But delayed initialization is almost never the right answer. Maybe I don't have a complete understanding of how this class is used in the algorithm.

Reply:
this is 2/3 remaining outstanding comments. as explained by Anders, it does not seem better to use null pointers that involve creating and deleting large number of objects every event?